content/docs/04.5-devx/05-app-profile.md

```diff
@@ -18,21 +18,101 @@ import Tooltip from "shared/components/ui/Tooltip";
 
+- **Helm**: Palette provides out-of-box Helm registries and allows you to add registries. For more information, visit [Palette Helm Registry](/devx/registries/helm-registry#palettehelmregistry).
 
+- **Manifest**: You can construct App Profile layers using raw manifests to provision Kubernetes resources that are unavailable in Palette or Helm Charts. Pack Manifests provide a pass-through mechanism to orchestrate Kubernetes resources in a cluster. For example, specific integrations may require creation of Secrets or Custom Resource Definitions (CRDs). To achieve this, you can attach a Manifest file to the layer.
 
+<InfoBox>
+We encourage you to securely distribute credentials using secrets. You can achieve this by constructing manifests to consume parameters that use references. You can also add a manifest to create your Kubernetes secrets using macros to inject the values. In the App layer, set the environment variables using a Secret reference.
 
+**Example**
 
+```
+apiVersion: v1
+kind: Secret
+metadata:
+  name: mysecret
+  namespace: {{.spectro.system.apptier.NAMESPACE}}
+data:
+  
+  namespace: "{{.spectro.system.apptier.NAMESPACE}}" # Resolves to a value of the variable defined in App Profile tier parameters.yaml file.
+
+  password: "{{.spectro.app.$appDepName-mongodb.PASSWORD}}" # To refer the tier output variables of the top tiers.
 	
+```
+```
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: myweb
+  namespace: mywebnamespace
+  ...
+spec:
+  template:
+  ...
+      containers:
+        - env:
+            - name: PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: mysecret
+                  key: password
+```
+
+</InfoBox>
+
+<br />
+
+<WarningBox>
+    When adding services to App Profiles using manifest files, specify a namespace name to which the resource belongs.
+</WarningBox> 
+
+## Messaging System Services
+
+These Messaging System services are available for client-server data exchange and monitoring:
+
+- **NATS**
+- **Kafka**
+<br />
+
+## Object Storage Services
+
+**MinIO** is available as a distributed object storage system.
+<br />
+
+
+## Database Services
+
+These Database Services are available: 
+
+- **MongoDB**
+- **MySQL** 
+- **PostgreSQl** 
+- **Redis**
+
+
+<InfoBox>
+
+  Database services have output variables that follow a usage hierarchy. Variable values from the first services you add, which become the first layer in the App Profile stack, can pass to services you add next, which appear higher in the stack. Variable values _cannot_ pass from the top service layers down.
```

Comment:
Not only database services, but similar is possible for messaging, object storage etc

Reply:
Revised the first sentence to say "Variables used in services follow a usage hierarchy. "